components/popover/index.js

```diff
@@ -35,6 +35,7 @@ const ARROW_OFFSET = 20;
  * @type {String}
  */
 const SLOT_NAME = 'Popover';
+const isMobile = () => window.innerWidth < 782;
```

Comment:
This is one more place to sync with our scss variables. I think for now it is ok but this logic could benefit from https://github.com/WordPress/gutenberg/pull/3331 where we add a synchronization with our scss variables, and from https://github.com/WordPress/gutenberg/pull/3332 where we add responsive-redux making is-mobile being a prop the component receives.

Reply:
The problem is, the `Popover` is a generic component outside the `editor` module. I thought about factorizing this but we can't assume any context in the `components` module.

Reply:
Right, I would say give this it would be better to simplify the component and just receive a showFullScreen and than is up the callers to decide if is mobile and the enable this option (maybe using responsive redux) or they don't enable this option. The downside is we would need to activate this option manually in the cases we want.

Reply:
That's an option but I worry about having the component's user compute the `isMobile` on its own. I think this would create more duplication than it's supposed to avoid especially for third-party code or WP code no in the `editor` module.

I personally don't mind the small duplication here to make it generic enough.

Reply:
Yes, that is true it would require more effort from callers. In that case, another option would be instead of expandOnMobile we receive a prop showFullScreen that defaults to our isMobile function (If no value is passed). The advantage is that callers are able to use their own logic and their own breakpoint for mobile.